tasks/Task.cpp

```diff
@@ -135,37 +149,55 @@ void Task::writeIMUSample(std::unique_ptr<uint8_t[]> const& pkt_buffer)
 
 void Task::convertDataAndWriteOutput(LidarScan& scan)
 {
-
-    base::samples::DepthMap depth_map;
     auto time = base::Time::now();
-    depth_map.time = time;
-    depth_map.timestamps.push_back(time);
+    m_depth_map.time = time;
+    m_depth_map.timestamps.push_back(time);
+
+    auto range = scan.field(sensor::ChanField::RANGE);
+    auto range_destaggered =
+        destagger<uint32_t>(range, m_metadata.format.pixel_shift_by_row);
 
-    depth_map.vertical_projection = base::samples::DepthMap::PROJECTION_TYPE::POLAR;
-    depth_map.horizontal_projection = base::samples::DepthMap::PROJECTION_TYPE::POLAR;
+    auto reflectivity_destaggered = getReflectivity(scan);
 
     auto width = m_metadata.format.columns_per_frame;
     auto height = m_metadata.format.pixels_per_column;
-
-    depth_map.horizontal_interval.push_back(M_PI * 2.0);
-    depth_map.horizontal_interval.push_back(0);
-
-    depth_map.vertical_interval.push_back(-(m_vertical_fov / 2) * M_PI / (180.0));
-    depth_map.vertical_interval.push_back((m_vertical_fov / 2) * M_PI / (180.0));
-
-    depth_map.vertical_size = m_metadata.format.pixels_per_column;
-    depth_map.horizontal_size = m_metadata.format.columns_per_frame;
-
-    auto img = scan.field(sensor::ChanField::RANGE);
-    auto data = destagger<uint32_t>(img, m_metadata.format.pixel_shift_by_row);
-
-    depth_map.distances.resize(data.rows() * data.cols());
+    m_depth_map.distances.clear();
+    m_depth_map.remissions.clear();
+    m_depth_map.distances.resize(range_destaggered.rows() * range_destaggered.cols());
+    m_depth_map.remissions.resize(range_destaggered.rows() * range_destaggered.cols());
     for (unsigned int h = 0; h < height; h++) {
         for (unsigned int w = 0; w < width; w++) {
-            depth_map.distances[h * width + w] = static_cast<double>(data(h, w)) / 1000.0;
+            unsigned int index = h * width + w;
+            m_depth_map.distances[index] =
+                static_cast<double>(range_destaggered(h, w)) / 1000.0;
+            if (reflectivity_destaggered(h, w) <= 100) {
+                m_depth_map.remissions[index] =
+                    static_cast<double>(reflectivity_destaggered(h, w) / 100.0);
```

Comment:
When possible, we should use multiplication over division. Especially in computer intensive tasks like this one.

source: https://stackoverflow.com/a/17883577

```suggestion
                static_cast<double>(range_destaggered(h, w)) * 1e-3;
            if (reflectivity_destaggered(h, w) <= 100) {
                m_depth_map.remissions[index] =
                    static_cast<double>(reflectivity_destaggered(h, w) * 1e-2);
```

Reply:
good call